api/api/V1/Recording.ts

```diff
@@ -1279,6 +1280,75 @@ export default (app: Application, baseUrl: string) => {
     }
   );
 
+  /**
+   * @api {get} /api/v1/recordings/track-tags/count Get track tag counts
+   * @apiName GetTrackTagCounts
+   * @apiGroup Tracks
+   * @apiDescription Fetches track tag counts grouped by tag, group, station, and user.
+   *                 Filters can be applied to narrow down the results.
+   *
+   * @apiUse V1UserAuthorizationHeader
+   *
+   * @apiParam (Query) {String} [type=thermalRaw] Type of recordings (thermalRaw/audio).
+   * @apiParam (Query) {Boolean} [includeAI=false] Include AI tags.
+   * @apiParam (Query) {String} [view-mode] View mode. Allows a super-user to view as a regular user.
+   * @apiParam (Query) {String[]} [exclude] Exclude specified tags.
+   * @apiParam (Query) {Number} [offset] Zero-based page number. Use '0' to get the first page.
+   * @apiParam (Query) {Number} [limit] Max number of records to be returned.
+   * @apiParam (Query) {Number} [groupId] Optional group ID to filter results by a specific group.
+   *
+   * @apiUse V1ResponseSuccess
+   * @apiSuccess {Object[]} rows List of track tag counts.
+   * @apiSuccess {String} rows.label Name of the track tag.
+   * @apiSuccess {Number} rows.userId User ID of the user who tagged or AI.
+   * @apiSuccess {Object} rows.group Group details.
+   * @apiSuccess {Number} rows.group.id ID of the group.
+   * @apiSuccess {String} rows.group.name Name of the group.
+   * @apiSuccess {Object} rows.station Station details.
+   * @apiSuccess {Number} rows.station.id ID of the station.
+   * @apiSuccess {String} rows.station.name Name of the station.
+   * @apiSuccess {Object} rows.device Device details.
+   * @apiSuccess {Number} rows.device.id ID of the device.
+   * @apiSuccess {String} rows.device.name Name of the device.
+   *
+   * @apiUse V1ResponseError
+   */
+  app.get(
+    `${apiUrl}/track-tags/count`,
+    extractJwtAuthorizedUser,
+    validateFields([
+      query("exclude").default([]).optional().isArray(),
+      query("includeAI").default(false).isBoolean(),
+      integerOf(query("offset")).optional(),
+      integerOf(query("limit").optional()),
+      query("type")
+        .default("thermalRaw")
+        .optional()
+        .isIn(Object.values(RecordingType)),
+      query("view-mode").optional().equals("user"),
+      integerOf(query("groupId")).optional(), // Added validation for groupId
```

Comment:
A style note:  at a glance at this API I can't tell that you're validating that the request user belongs to the group (in the non-super-user case).  I dug through the getTrackTagsCount and I see that it does in fact add the correct constraint to the SQL that it generates – but that's an internal implementation detail.  Generally speaking I think it's good to make those contracts obvious as part of the API validation boilerplate i.e. using `fetchAuthorizedRequiredGroupById` – *even* though that's at the expense of a tiny amount of efficiency.  

As some background, in the distant past we had validation kind of threaded through the entire api/model ad-hoc, and it was extremely difficult to work out by looking at an API end-point whether validation existed, and if so, where in the call-stack.  It also took a long time to untangle, and I'd like to see us not go back there!